boggle.js

```diff
@@ -7,25 +9,23 @@ function newBoggleBoard() {
   ];
 }
 
-/**
- * Prints out a boggle board.
- */
 function printBoggleBoard(boggleBoard) {
   for (let row of boggleBoard) {
     console.log(row.join(' '));
   }
 }
 
-/**
- * Shake a boggle board and fill it with letters.
- */
-function shake(boggleBoard) {
-  // This is your job. :)
-  return boggleBoard;
-}
 
+function shake(board) {
+ for (a=0; a<=15; a++){
+   board[a] = alphabet[Math.floor(Math.random()*26)];
```

Comment:
Consider making a function that takes a list of letters as input and gives you back one of them randomly, e.g.,

```javascript
board[a] = randomLetter(alphabet);
```

All `randomLetter` needs to know is the elements of `alphabet` and the length of `alphabet`, so passing in `alphabet` provides everything it needs to do its job.

Reply:
Actually...does this work? I don't think this works. `board` is an array of arrays. `board.length` will only be `4`. You need to iterate both over rows and columns.

Reply:
You're (unintentionally) changing the structure of `board` from a list of four, 4-element arrays to a list of 16 elements.